cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/RunnableTaskLauncher.java

```diff
@@ -27,10 +27,14 @@
  * RunnableTaskLauncher launches a {@link RunnableTask} by loading its class and calling its run method.
  */
 public class RunnableTaskLauncher {
-  private final CConfiguration cConf;
+  private final Injector injector;
 
   public RunnableTaskLauncher(CConfiguration cConf) {
-    this.cConf = cConf;
+    injector = Guice.createInjector(new RunnableTaskModule(cConf));
+  }
+
+  public RunnableTaskLauncher(Injector injector) {
```

Comment:
can you add some comments on when this is used and how it is different than the other constructor?